library/src/main/java/com/chuckerteam/chucker/internal/support/FileFactory.kt

```diff
@@ -3,5 +3,5 @@ package com.chuckerteam.chucker.internal.support
 import java.io.File
 
 internal interface FileFactory {
-    fun create(): File
+    fun create(filename: String = ""): File
```

Comment:
```suggestion
    fun create(filename: String): File
```

Reply:
@cortinico any reason why default parameter should be removed from the `FileFactory` interface? the reason why I added was API can be used as it was earlier `fileFactory.create()` , in addition to that people can pass filename if required(filename being optional) instead of mandatorily passing an empty string as `fileFactory.create("")`

Reply:
I generally feel that having a method `fun create(filename: String = ""): File` in an interface is a bad API design practice. What you're saying here is that you offer a _create_ method that will return a File given a filename that you could not provide, if you don't provide the filename is going to be `""`.

What is happening is that in your implementation you're actually ignoring the parameter if is `""` and you're there defining the real _default value_. So I suggest you either:

* Convert `FileFactory` to an `abstract class`. Move up the `AtomicLong` and have a single method like: `abstract fun create(filename: String = "chucker-${uniqueIdGenerator.getAndIncrement()}"): File`. There you actually be backward compatible and the intent would be clearer.

* Offer two `create` method in the interface: a `create()` (for backward compatibility and a `create(filename: String)` with no default parameter.